src/content/docs/fr/guides/cms/hashnode.mdx

```diff
@@ -0,0 +1,352 @@
+                }),
+                edges: z.array(z.object({
+                    node: PostSchema,
+                })),
+            }),
+        }),
+    })
+
+    export const PostDataSchema = z.object({
+        publication: z.object({
+            title: z.string(),
+            post: PostSchema,
+        }),
+    })
+
+    export type Post = z.infer<typeof PostSchema>
+    export type AllPostsData = z.infer<typeof AllPostsDataSchema>
+    export type PostData = z.infer<typeof PostDataSchema>
+
+    ```
+</Steps>
+
+### Affichage d'une liste d'articles
+
+La recherche via `getAllPosts()` renvoie un tableau d'objets contenant les propriétés de chaque article telles que :
+- `title` - le titre de l'article
+- `brief` - l'affichage HTML du contenu de l'article
+- `coverImage.url` - l'URL source de l'image vedette de l'article
+- `slug` - le nom de l'article.
+
+Utilisez le tableau `posts` renvoyé par la recherche pour afficher une liste d'articles de blog sur la page.
+
+
+```astro title="src/pages/index.astro"
+---
+import { getAllPosts } from '../lib/client';
+
+const data = await getAllPosts();
+const allPosts = data.publication.posts.edges;
+
+---
+
+<html lang="en">
+    <head>
+        <title>Astro + Hashnode</title>
+    </head>
+    <body>
+
+        {
+            allPosts.map((post) => (
+                <div>
+                    <h2>{post.node.title}</h2>
+                    <p>{post.node.brief}</p>
+                    <img src={post.node.coverImage.url} alt={post.node.title} />
+                    <a href={`/post/${post.node.slug}`}>Lire plus</a>
+                </div>
+            ))
+        }
+    </body>
+</html>
+```
+
+### Générer des pages
+
+<Steps>
+1. Créez la page `src/pages/post/[slug].astro` pour [générer dynamiquement une page](/fr/guides/routing/#routes-dynamiques) pour chaque post.
+
+    <FileTree title="Project Structure">
+    - src/
+    - lib/
+        - client.ts
+        - schema.ts
+      - pages/
+        - index.astro
+        - post/
+          - **[slug].astro**
+    - astro.config.mjs
+    - package.json
+    </FileTree>
+
+2. Importez et utilisez `getAllPosts()` et `getPost()` pour récupérer les données de Hashnode et générer des routes de pages individuelles pour chaque post.
+
+    ```astro title="src/pages/post/[slug].astro"
+    ---
+    import { getAllPosts, getPost } from '../../lib/client';
+
+
+    export async function getStaticPaths() {
+      const data = await getAllPosts();
+      const allPosts = data.publication.posts.edges;
+      return allPosts.map((post) => {
+        return {
+          params: { slug: post.node.slug },
+        }
+      })
+    }
+    const { slug } = Astro.params;
+    const post = await getPost(slug);
+
+    ---
+    ```
+
+3. Créez le modèle pour chaque page en utilisant les propriétés de chaque objet `post`. L'exemple ci-dessous montre le titre de l'article et l'heure de lecture, puis le contenu complet de l'article :
```

Comment:
It's the estimated reading time; I understand "heure de lecture" like the time when the article is read. So I think "temps de lecture" is more suitable.
```suggestion
3. Créez le modèle pour chaque page en utilisant les propriétés de chaque objet `post`. L'exemple ci-dessous montre le titre de l'article et le temps de lecture, puis le contenu complet de l'article :
```